dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScope.kt

```diff
@@ -29,6 +31,43 @@ interface FeatureScope {
         callback: (DatadogContext, EventBatchWriter) -> Unit
     )
 
+    /**
+     * Write data to the datastore.
+     *
+     * @param T datatype of the data to write to the datastore.
+     * @param dataStoreFileName name of the datastore file as there could be multiple such files per feature.
```

Comment:
I think the rationale on ios related to how difficult it would be to handle corruption in a large file vs in multiple smaller ones, and avoiding the complexity of seeking/updating records